File: test/java/com/marsrovers/units/RoverTest.java
package com.marsrovers.units;

import com.marsrovers.location.Position;
import org.junit.Test;

import static com.marsrovers.location.Orientation.*;
import static org.junit.Assert.assertEquals;

public class RoverTest {

    private final String roverId = "a";
    private final Rover rover = new Rover(roverId, NORTH, new Position(1,1));


    @Test
    public void roverShouldMoveOrientationLeft() throws Exception {

        Rover newRover = rover.left();

        assertEquals(WEST, newRover.getOrientation());

    }

    @Test
    public void locationShouldMoveOrientationRight() throws Exception {

        Rover newRover = rover.right();

        assertEquals(EAST, newRover.getOrientation());

    }

    @Test
    public void locationShouldMovePositionUp() throws Exception {

        Rover newRover = rover.move();

        assertEquals(2, newRover.getPosition().getY());

    }

    @Test
    public void locationShouldMovePositionDown() throws Exception {

        Rover rover = new Rover(roverId, SOUTH, new Position(1,1));

        Rover newRover = rover.move();

        assertEquals(0, newRover.getPosition().getY());

    }

    @Test
    public void toStringShouldBeCorrect() throws Exception {
        String asString = rover.toString();

        assertEquals("a " + rover.getPosition().toString() + " " + rover.getOrientation().toString(), asString);

    }
}

File: test/java/com/marsrovers/units/RoverTest.kt
package com.marsrovers.units

import com.marsrovers.location.Orientation
import com.marsrovers.location.Position
import org.junit.Test
import org.junit.Assert.assertEquals

class RoverTest {

    private val roverId = "a"
    private val rover = Rover(roverId, Orientation.NORTH, Position(1, 1))


    @Test
    @Throws(Exception::class)
    fun roverShouldMoveOrientationLeft() {

        val newRover = rover.left()

        assertEquals(Orientation.WEST, newRover.orientation)

    }

    @Test
    @Throws(Exception::class)
    fun locationShouldMoveOrientationRight() {

        val newRover = rover.right()

        assertEquals(Orientation.EAST, newRover.orientation)

    }

    @Test
    @Throws(Exception::class)
    fun locationShouldMovePositionUp() {

        val newRover = rover.move()

        assertEquals(2, newRover.position.y.toLong())

    }

    @Test
    @Throws(Exception::class)
    fun locationShouldMovePositionDown() {

        val rover = Rover(roverId, Orientation.SOUTH, Position(1, 1))

        val newRover = rover.move()

        assertEquals(0, newRover.position.y.toLong())

    }

    @Test
    @Throws(Exception::class)
    fun toStringShouldBeCorrect() {
        val asString = rover.toString()

        assertEquals("a " + rover.position.toString() + " " + rover.orientation.toString(), asString)

    }
}